test/conformance/envvars_test.go

```diff
@@ -0,0 +1,77 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+)
+
+//TestShouldEnvVars verifies environment variables that are declared as "SHOULD be set" in runtime-contract
+func TestShouldEnvVars(t *testing.T) {
+	logger := logging.GetContextLogger("TestShouldEnvVars")
+	var names test.ResourceNames
+	resp, err := fetchEnvInfo(t, logger, test.EnvImageEnvVarsPath, &names)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var respValues ShouldEnvvars
+	if err := json.Unmarshal(resp, &respValues); err != nil {
+		t.Fatalf("Failed to unmarshall response : %v", err)
+	}
+
+	expectedValues := ShouldEnvvars {
+		Service: names.Service,
+		Configuration : names.Config,
+		Revision : names.Revision,
+	}
+	if respValues != expectedValues {
+		t.Fatalf("Received response failed to match execpted response. Received: %v Expected: %v", respValues, expectedValues)
+	}
+}
+
+//TestMustEnvVars verifies environment variables that are declared as "MUST be set" in runtime-contract
+func TestMustEnvVars(t *testing.T) {
+	logger := logging.GetContextLogger("TestMustEnvVars")
+	var names test.ResourceNames
+	resp, err := fetchEnvInfo(t, logger, test.EnvImageEnvVarsPath, &names)
```

Comment:
this resp is the same as what's in TestShouldEnvVars?

Reply:
The response from the environment test image is retrieving all the environment variables defined inside the container. And the type of object used for json.Unmarshall is determining what are environment variables of interest, in this case type MustEnvvars. The advantage of this is the next time we add/delete any environment variable from the list of SHOULD or MUST we would only have to change it in one place which is runtime_contract_types.go and everything else would just work.